src/dmd/dinterpret.d

```diff
@@ -6095,6 +6095,30 @@ public:
             se = (cast(ClassReferenceExp)ex).value;
             i = (cast(ClassReferenceExp)ex).findFieldIndexByName(v);
         }
+        else if (ex.op == TOK.typeid_)
+        {
+            auto tie = cast(TypeidExp) ex;
+            if (v.ident == Identifier.idPool("name"))
+            {
+                if (auto t = isType(tie.obj))
+                {
+                    auto sym = t.toDsymbol(null);
+                    auto ident = (sym ? sym.ident : null);
+                    if (ident)
+                    {
+                        result = new StringExp(e.loc, ident.toString());
+                        result.expressionSemantic(null);
+                        return ;
+                    }
+                }
+            }
+            // We get here if we could not resolve the name or we were not asked for the name
+            {
```

Comment:
The scope block is unnecessary.

Reply:
The block is the target of multiple failing branches, I prefer the error-code to be grouped that way.
I plan to extend this functionality and that format is the friendly to extension.

Reply:
Simply grouping the instructions in a comment + a paragraph (preceded and succeed by a blank line) is more then enough from a readability standpoint and it uses less lines of code.